src/realm/query_expression.hpp

```diff
@@ -3883,6 +3986,22 @@ class Compare : public Expression {
     {
         m_left->set_base_table(table);
         m_right->set_base_table(table);
+        if (std::is_same<TCond, Equal>::value && m_left_is_const && m_right->has_search_index()) {
```

Comment:
Do we know that this will be triggered if the query is re-evaluated?

Reply:
Yes, the init function is called as part of both Query::find and Query::find:all